### ISSUE TITLE
Sort locations in the booking modal

### ISSUE THREAD

**Issue by oeed** [CovidEngine/vaxxnz#54]
When you open the booking modal by clicking a date on the calendar, let users sort by:
- distance
- number of slots available

**Comment by Francismb** [comment]
It looks like it is already sorted by distance... I think what would be very beneficial is to be able to select the time that you want to get a vaccine and it shows you the nearest locations that can provide a vaccine at that time?

**Comment by oeed** [comment]
Yeah this came up on the Discord last night. You raise a good point though, number of lots available isn't really helpful on it's own.

Maybe we could have a way of filtering out/dimming times? i.e. let users pick 'morning', 'midday', etc. and times outside that are hidden/dimmed? Thoughts @waltzaround ?

**Comment by Francismb** [comment]
I think that is a much better solution! Currently I also feel the user needs to scroll quite a bit in order to see all available times for all locations on that date.

**Comment by waltzaround** [comment]
Agreed,  would be worth exploring how we aggregate and sort by time here

**Comment by waltzaround** [comment]
so as opposed to how we do it now

site
 - time slot
 - time slot 
 - time slot 
 
 
 
 Maybe a toggle that changes it to
 
 time
 - site
 - site 
 - site

**Comment by noway** [comment]
We need a design for this :heart:

**Comment by Francismb** [comment]
I have started work on a quick select at the top. User will be able to select 'Morning', 'Midday', 'Afternoon', 'Evening' and that will limit the times we show and possibly the amount of locations also. I believe the idea of vaxx.nz is to show times instead of locations first which is what the government booking website got wrong right? I think we should continue heading in that direction. Keen on feedback though!

**Comment by noway** [comment]
@Francismb Nice. I personally like the idea. Not only sorting, but filtering too. 

Let's do it 🚀
